webapp/packages/plugin-data-viewer/src/TableViewer/TableFooter/TableFooterMenu/TableFooterMenuService.ts

```diff
@@ -191,7 +191,15 @@ export class TableFooterMenuService {
             break;
           }
           case ACTION_SAVE:
-            model.save().catch(() => {});
+            model
+              .save()
+              .then(() => {
+                //TODO: Remove this when we have virtual keys. We need to refresh the data in tables without a primary key to avoid UI glitch #5140.
+                if (!model.hasElementIdentifier(resultIndex)) {
+                  model.source.refreshData();
+                }
+              })
+              .catch(() => {});
```

Comment:
if possible, it's better to do it in `ResultSetDataSource` by overriding `saveData` method

Reply:
Why do you think it's better? I thought about that but decided to keep it here because refreshing the table after saving  is just a temporary side effect and has nothing in common with actual saving process.

Reply:
Moved to datasource